src/hand.rs

```diff
@@ -0,0 +1,100 @@
+use super::bag::TileBag;
+
+pub struct Hands {
+    hands: Vec<Vec<char>>,
+    bag: TileBag,
+}
+
+impl Hands {
+    pub fn new(player_count: usize, capacity: usize, mut bag: TileBag) -> Self {
+        let mut hands = Vec::new();
+        for player in 0..player_count {
+            hands.push(Vec::new());
+            for _ in 0..capacity {
+                hands[player].push(bag.draw_tile());
+            }
+        }
+        Self { hands, bag }
+    }
+
+    pub fn use_tile(&mut self, player: usize, tile: char) -> Result<(), &str> {
+        if self.hands.len() <= player {
+            return Err("Invalid player"); // TODO: say which player is wrong
+        }
+
+        let index = self.hands[player].iter().position(|t| t == &tile);
+        match index {
+            None => Err("Player doesn't have that tile"), // TODO: say which player and tile
+            Some(index) => {
+                self.hands[player][index] = self.bag.draw_tile();
+                Ok(())
+            }
+        }
+    }
```

Comment:
Firmly a "alt way to write it" that involves less self.hands

```suggestion
    pub fn use_tile(&mut self, player: usize, tile: char) -> Result<(), &str> {
        if let Some(player) = self.hands.get_mut(player) {
            let index = player.iter().position(|t| t == &tile);
            match index {
                None => Err("Player doesn't have that tile"), // TODO: say which player and tile
                Some(index) => {
                    player[index] = self.bag.draw_tile();
                    Ok(())
                }
            }
        } else {
            Err("Invalid player") // TODO: say which player is wrong
        }
    }
```

Reply:
I also like that the whole function body is just one branching expression. Makes it easier to look at a returned value and visualise the conditions that led to it - you just look up the tree rather than having to read through earlier statements as well.